<a href="https://colab.research.google.com/github/sejalvyas13/Basic-CYK-Parser/blob/master/MTNT_finetuned_on_IWSLT.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Fine tuning MTNT based on trained results of IWSLT. Thus training for MTNT starts from epoch 19 or something

In [ ]:
!git clone https://github.com/abdulrafae/nmt_tutorial.git

Cloning into 'nmt_tutorial'...
remote: Enumerating objects: 88, done.
remote: Counting objects: 100% (88/88), done.
remote: Compressing objects: 100% (87/87), done.
remote: Total 88 (delta 46), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (88/88), done.


In [ ]:
cd nmt_tutorial/

/content/nmt_tutorial


In [ ]:
!pip install fairseq


     |████████████████████████████████| 1.7MB 9.1MB/s 
     |████████████████████████████████| 61kB 7.0MB/s 
     |████████████████████████████████| 133kB 48.8MB/s 
     |████████████████████████████████| 112kB 35.9MB/s 
     |████████████████████████████████| 645kB 33.2MB/s 
  Created wheel for antlr4-python3-runtime: filename=antlr4_python3_runtime-4.8-cp37-none-any.whl size=141231 sha256=bcdfc3962a402d76463fb459b2a8f6dbaf48eb2b698da4917f309d46786f8b9a
  Stored in directory: /root/.cache/pip/wheels/e3/e2/fa/b78480b448b8579ddf393bebd3f47ee23aa84c89b6a78285c8
Successfully built antlr4-python3-runtime
  Found existing installation: PyYAML 3.13
    Uninstalling PyYAML-3.13:
      Successfully uninstalled PyYAML-3.13


In [ ]:
!git clone https://github.com/moses-smt/mosesdecoder.git


Cloning into 'mosesdecoder'...
remote: Enumerating objects: 148070, done.
remote: Counting objects: 100% (498/498), done.
remote: Compressing objects: 100% (206/206), done.
remote: Total 148070 (delta 315), reused 433 (delta 289), pack-reused 147572
Receiving objects: 100% (148070/148070), 129.86 MiB | 26.90 MiB/s, done.
Resolving deltas: 100% (114341/114341), done.


In [ ]:
!git clone https://github.com/rsennrich/subword-nmt.git


Cloning into 'subword-nmt'...
remote: Enumerating objects: 580, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 580 (delta 0), reused 0 (delta 0), pack-reused 576
Receiving objects: 100% (580/580), 237.41 KiB | 8.48 MiB/s, done.
Resolving deltas: 100% (349/349), done.


In [ ]:
!git clone https://github.com/pytorch/fairseq

Cloning into 'fairseq'...
remote: Enumerating objects: 27747, done.
remote: Counting objects: 100% (168/168), done.
remote: Compressing objects: 100% (92/92), done.
remote: Total 27747 (delta 87), reused 118 (delta 76), pack-reused 27579
Receiving objects: 100% (27747/27747), 11.59 MiB | 26.14 MiB/s, done.
Resolving deltas: 100% (20902/20902), done.


In [ ]:
cd fairseq


/content/nmt_tutorial/fairseq


In [ ]:
!pip install -r requirements.txt
!python setup.py build develop

ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.7
creating build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/generate.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/hydra_train.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/preprocess.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/train.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/score.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/eval_lm.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/__init__.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/validate.py -> build/lib.linux-x86_64-3.7/fairseq_cli
copying fairseq_cli/interactive.py -> build/lib.linux-x86_64-3.7/fairseq_cli
creating build/lib.linux-x86_64-3.7/fairseq
copying fairseq/iterative_refinement_generator.py -> build/lib

In [ ]:
!pip install --editable ./

Obtaining file:///content/nmt_tutorial/fairseq
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
  Found existing installation: fairseq 0.10.2
    Uninstalling fairseq-0.10.2:
      Successfully uninstalled fairseq-0.10.2
  Running setup.py develop for fairseq


In [ ]:
cd ..

/content


In [ ]:
cd nmt_tutorial/

/content/nmt_tutorial


In [ ]:
!bash prepare_data.sh


In [ ]:
cd /content


/content


In [ ]:
!echo $PYTHONPATH

import os

os.environ['PYTHONPATH'] += ":/content/nmt_tutorial/fairseq/" #path to your fairseq

!echo $PYTHONPATH

/env/python:/content/nmt_tutorial/fairseq/:/content/nmt_tutorial/fairseq/
/env/python:/content/nmt_tutorial/fairseq/:/content/nmt_tutorial/fairseq/:/content/nmt_tutorial/fairseq/


In [ ]:
cd nmt_tutorial/

/content/nmt_tutorial


In [ ]:
!rm -rf data-bin/

In [ ]:
!wc iwslt17.tokenized.fr-en/train.fr

  140882  3054542 16777216 iwslt17.tokenized.fr-en/train.fr


In [ ]:
!mkdir temp

In [ ]:
!mv iwslt17.tokenized.fr-en/test.en  temp/test.en

In [ ]:
!mv iwslt17.tokenized.fr-en/test.fr  temp/test.fr
!mv iwslt17.tokenized.fr-en/train.en  temp/train.en
!mv iwslt17.tokenized.fr-en/train.fr  temp/train.fr
!mv iwslt17.tokenized.fr-en/valid.en  temp/valid.en
!mv iwslt17.tokenized.fr-en/valid.fr  temp/valid.fr

In [ ]:
!cat temp/train.fr | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > iwslt17.tokenized.fr-en/train.tok.fr
!cat temp/valid.fr | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > iwslt17.tokenized.fr-en/valid.tok.fr
!cat temp/train.en | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > iwslt17.tokenized.fr-en/train.tok.en
!cat temp/valid.en | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > iwslt17.tokenized.fr-en/valid.tok.en
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < iwslt17.tokenized.fr-en/train.tok.fr > iwslt17.tokenized.fr-en/train.fr-en.tc.fr
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < iwslt17.tokenized.fr-en/valid.tok.fr > iwslt17.tokenized.fr-en/valid.fr-en.tc.fr
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < iwslt17.tokenized.fr-en/train.tok.en > iwslt17.tokenized.fr-en/train.fr-en.tc.en
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < iwslt17.tokenized.fr-en/valid.tok.en > iwslt17.tokenized.fr-en/valid.fr-en.tc.en
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < iwslt17.tokenized.fr-en/train.fr-en.tc.en > iwslt17.tokenized.fr-en/train.en
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < iwslt17.tokenized.fr-en/train.fr-en.tc.fr > iwslt17.tokenized.fr-en/train.fr
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < iwslt17.tokenized.fr-en/valid.fr-en.tc.en > iwslt17.tokenized.fr-en/valid.en
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < iwslt17.tokenized.fr-en/valid.fr-en.tc.fr > iwslt17.tokenized.fr-en/valid.fr

Tokenizer Version 1.1
Language: en
Number of threads: 8
Tokenizer Version 1.1
Language: en
Number of threads: 8
Tokenizer Version 1.1
Language: en
Number of threads: 8
Tokenizer Version 1.1
Language: en
Number of threads: 8
Error: invalid line 2 in BPE codes file: 
The line should exist of exactly two subword units, separated by whitespace
Error: invalid line 2 in BPE codes file: 
The line should exist of exactly two subword units, separated by whitespace
Error: invalid line 2 in BPE codes file: 
The line should exist of exactly two subword units, separated by whitespace
Error: invalid line 2 in BPE codes file: 
The line should exist of exactly two subword units, separated by whitespace


In [ ]:
!wc iwslt17.tokenized.fr-en/train.fr

0 0 0 iwslt17.tokenized.fr-en/train.fr


In [ ]:
!bash train_fconv.sh


2021-04-30 15:01:43 | INFO | fairseq_cli.preprocess | Namespace(align_suffix=None, alignfile=None, all_gather_list_size=16384, azureml_logging=False, bf16=False, bpe=None, cpu=False, criterion='cross_entropy', dataset_impl='mmap', destdir='data-bin/iwslt17_fr_en', empty_cache_freq=0, fp16=False, fp16_init_scale=128, fp16_no_flatten_grads=False, fp16_scale_tolerance=0.0, fp16_scale_window=None, joined_dictionary=False, log_file=None, log_format=None, log_interval=100, lr_scheduler='fixed', memory_efficient_bf16=False, memory_efficient_fp16=False, min_loss_scale=0.0001, model_parallel_size=1, no_progress_bar=False, nwordssrc=-1, nwordstgt=-1, only_source=False, optimizer=None, padding_factor=8, plasma_path='/tmp/plasma', profile=False, quantization_config_path=None, reset_logging=False, scoring='bleu', seed=1, simul_type=None, source_lang='fr', srcdict=None, suppress_crashes=False, target_lang='en', task='translation', tensorboard_logdir=None, testpref=None, tgtdict=None, threshold_loss_

In [ ]:
!wget https://github.com/pmichel31415/mtnt/releases/download/v1.1/MTNT.1.1.tar.gz

--2021-04-30 16:32:05--  https://github.com/pmichel31415/mtnt/releases/download/v1.1/MTNT.1.1.tar.gz
Resolving github.com (github.com)... 192.30.255.112
Connecting to github.com (github.com)|192.30.255.112|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github-releases.githubusercontent.com/125262764/1b3bee80-1e45-11e9-90db-a9db0f519608?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210430%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210430T163206Z&X-Amz-Expires=300&X-Amz-Signature=dcb7ec07e265b3843700d7711ab0b99708ed0526a5fcbe4cd05a764670ea3859&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=125262764&response-content-disposition=attachment%3B%20filename%3DMTNT.1.1.tar.gz&response-content-type=application%2Foctet-stream [following]
--2021-04-30 16:32:06--  https://github-releases.githubusercontent.com/125262764/1b3bee80-1e45-11e9-90db-a9db0f519608?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH

In [ ]:
!tar -xvzf MTNT.1.1.tar.gz

MTNT/
MTNT/README.md
MTNT/monolingual/
MTNT/monolingual/dev.en
MTNT/monolingual/dev.fr
MTNT/monolingual/dev.ja
MTNT/monolingual/dev.tok.en
MTNT/monolingual/dev.tok.fr
MTNT/monolingual/dev.tok.ja
MTNT/monolingual/train.en
MTNT/monolingual/train.fr
MTNT/monolingual/train.ja
MTNT/monolingual/train.tok.en
MTNT/monolingual/train.tok.fr
MTNT/monolingual/train.tok.ja
MTNT/split_tsv.sh
MTNT/test/
MTNT/test/test.en-fr.tsv
MTNT/test/test.en-ja.tsv
MTNT/test/test.fr-en.tsv
MTNT/test/test.ja-en.tsv
MTNT/train/
MTNT/train/train.en-fr.tsv
MTNT/train/train.en-ja.tsv
MTNT/train/train.fr-en.tsv
MTNT/train/train.ja-en.tsv
MTNT/valid/
MTNT/valid/valid.en-fr.tsv
MTNT/valid/valid.en-ja.tsv
MTNT/valid/valid.fr-en.tsv
MTNT/valid/valid.ja-en.tsv


In [ ]:
!bash MTNT/split_tsv.sh

Preprocessing

In [ ]:
!cat MTNT/train/train.fr-en.fr | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > MTNT/train/train.fr-en.tok.fr
!cat MTNT/valid/valid.fr-en.fr | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > MTNT/valid/valid.fr-en.tok.fr
!cat MTNT/valid/valid.fr-en.en | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > MTNT/valid/valid.fr-en.tok.en
!cat MTNT/train/train.fr-en.en | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > MTNT/train/train.fr-en.tok.en
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < MTNT/train/train.fr-en.tok.fr > MTNT/train/train.fr-en.tc.fr
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < MTNT/train/train.fr-en.tok.en > MTNT/train/train.fr-en.tc.en
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < MTNT/valid/valid.fr-en.tok.fr > MTNT/valid/valid.fr-en.tc.fr
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < MTNT/valid/valid.fr-en.tok.en > MTNT/valid/valid.fr-en.tc.en
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < MTNT/train/train.fr-en.tc.fr > MTNT/train/train.fr-en.bpe.fr
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < MTNT/train/train.fr-en.tc.en > MTNT/train/train.fr-en.bpe.en
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < MTNT/valid/valid.fr-en.tc.fr > MTNT/valid/valid.fr-en.bpe.fr
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < MTNT/valid/valid.fr-en.tc.en > MTNT/valid/valid.fr-en.bpe.en

Tokenizer Version 1.1
Language: en
Number of threads: 8
Tokenizer Version 1.1
Language: en
Number of threads: 8
Tokenizer Version 1.1
Language: en
Number of threads: 8
Tokenizer Version 1.1
Language: en
Number of threads: 8


In [ ]:
!wc MTNT/train/train.fr-en.bpe.en

  19161  955161 4403221 MTNT/train/train.fr-en.bpe.en


Preprocessing

In [ ]:
!python fairseq/fairseq_cli/preprocess.py --source-lang fr --target-lang en \
    --trainpref MTNT/train/train.fr-en.bpe --validpref MTNT/valid/valid.fr-en.bpe \
    --destdir data-bin/MTNT_fr_en \
    --workers 20 --srcdict data-bin/iwslt17_fr_en/dict.fr.txt --tgtdict data-bin/iwslt17_fr_en/dict.en.txt

2021-04-30 17:33:18 | INFO | fairseq_cli.preprocess | Namespace(align_suffix=None, alignfile=None, all_gather_list_size=16384, azureml_logging=False, bf16=False, bpe=None, cpu=False, criterion='cross_entropy', dataset_impl='mmap', destdir='data-bin/MTNT_fr_en', empty_cache_freq=0, fp16=False, fp16_init_scale=128, fp16_no_flatten_grads=False, fp16_scale_tolerance=0.0, fp16_scale_window=None, joined_dictionary=False, log_file=None, log_format=None, log_interval=100, lr_scheduler='fixed', memory_efficient_bf16=False, memory_efficient_fp16=False, min_loss_scale=0.0001, model_parallel_size=1, no_progress_bar=False, nwordssrc=-1, nwordstgt=-1, only_source=False, optimizer=None, padding_factor=8, plasma_path='/tmp/plasma', profile=False, quantization_config_path=None, reset_logging=False, scoring='bleu', seed=1, simul_type=None, source_lang='fr', srcdict='data-bin/iwslt17_fr_en/dict.fr.txt', suppress_crashes=False, target_lang='en', task='translation', tensorboard_logdir=None, testpref=None, 

In [ ]:
!python fairseq/train.py data-bin/MTNT_fr_en \
    --skip-invalid-size-inputs-valid-test \
	--lr 0.25 --clip-norm 0.1 --dropout 0.2 \
	--max-tokens 4000 \
	--arch fconv_iwslt_de_en \
    --criterion label_smoothed_cross_entropy \
    --optimizer nag --label-smoothing 0.1 \
    --lr-scheduler fixed --force-anneal 50 \
	--max-epoch 50 --patience 5 \
	--save-dir checkpoint/MTNT_fr_en \
	--restore-file checkpoint/iwslt17_fr_en/checkpoint_best.pt | tee log/MTNT_fr_en/train_fconv.out

tee: log/MTNT_fr_en/train_fconv.out: No such file or directory
2021-04-30 17:33:29 | INFO | fairseq_cli.train | {'_name': None, 'common': {'_name': None, 'no_progress_bar': False, 'log_interval': 100, 'log_format': None, 'log_file': None, 'tensorboard_logdir': None, 'wandb_project': None, 'azureml_logging': False, 'seed': 1, 'cpu': False, 'tpu': False, 'bf16': False, 'memory_efficient_bf16': False, 'fp16': False, 'memory_efficient_fp16': False, 'fp16_no_flatten_grads': False, 'fp16_init_scale': 128, 'fp16_scale_window': None, 'fp16_scale_tolerance': 0.0, 'min_loss_scale': 0.0001, 'threshold_loss_scale': None, 'user_dir': None, 'empty_cache_freq': 0, 'all_gather_list_size': 16384, 'model_parallel_size': 1, 'quantization_config_path': None, 'profile': False, 'reset_logging': False, 'suppress_crashes': False, 'use_plasma_view': False, 'plasma_path': '/tmp/plasma'}, 'common_eval': {'_name': None, 'path': None, 'post_process': None, 'quiet': False, 'model_overrides': '{}', 'results_path': N

In [ ]:
!cat MTNT/test/test.fr-en.fr | mosesdecoder/scripts/tokenizer/tokenizer.perl -threads 8 -a -l en > MTNT/test/test.fr-en.tok.fr
!mosesdecoder/scripts/recaser/truecase.perl -model iwslt17.tokenized.fr-en/truecase-model.fr < MTNT/test/test.fr-en.tok.fr > MTNT/test/test.fr-en.tc.fr
!subword-nmt/subword_nmt/apply_bpe.py -c iwslt17.tokenized.fr-en/code < MTNT/test/test.fr-en.tc.fr > MTNT/test/test.fr-en.bpe.fr

Tokenizer Version 1.1
Language: en
Number of threads: 8


In [ ]:
mkdir Output

mkdir: cannot create directory ‘Output’: File exists


In [ ]:
!fairseq-interactive data-bin/MTNT_fr_en/ \
    --skip-invalid-size-inputs-valid-test \
    --path checkpoint/MTNT_fr_en/checkpoint_last.pt \
    --buffer-size 2000 --batch-size 128 \
    --beam 5 --remove-bpe \
    --input MTNT/test/test.fr-en.bpe.fr | tee Output/MTNT.test.fr-en.en.sys

Streaming output truncated to the last 5000 lines.
W-22	0.012	seconds
H-22	-0.882133424282074	<unk> day the direction their said that <unk> <unk> before , this kind of <unk> <unk> <unk> <unk> to pass by a <unk> <unk> of public markets .
D-22	-0.882133424282074	<unk> day the direction their said that <unk> <unk> before , this kind of <unk> <unk> <unk> <unk> to pass by a <unk> <unk> of public markets .
P-22	-0.1958 -1.3698 -1.9966 -0.6194 -0.8316 -2.1514 -0.6677 -0.1912 -0.3756 -0.2915 -0.2736 -0.5417 -0.2842 -0.0310 -0.6023 -0.9080 -1.7309 -1.6659 -2.1481 -3.1332 -0.7549 -0.4540 -0.6897 -1.2861 -1.6483 -0.3434 -0.1104 -0.1637 -0.1218
S-23	<unk> u coup maintenant , quand ils font leur <unk> <unk> t <unk> j , c <unk> est une société privée qui <unk> pêche un type pour faire 60 <unk> nes avec la <unk> <unk> et le café .
W-23	0.014	seconds
H-23	-0.6621608138084412	<unk> now , when they do their <unk> <unk> t <unk> <unk> , it &apos;s a private society that <unk> <unk> a guy to make 60 <unk> 

In [ ]:
!grep ^H Output/MTNT.test.fr-en.en.sys | cut -f3- > Output/hypo.tok.sys


In [ ]:
!sed -e 's/<\/unk>//g' Output/hypo.tok.sys > Output/hypo.tok.sys1

In [ ]:
!awk '{gsub("<unk>", "");print}' Output/hypo.tok.sys > Output/hypo.tok.sys1

In [ ]:
!cat Output/hypo.tok.sys1 | mosesdecoder/scripts/recaser/detruecase.perl | mosesdecoder/scripts/tokenizer/detokenizer.perl -l en > Output/hypo.sys


Detokenizer Version $Revision: 4134 $
Language: en


In [ ]:
!sacrebleu --tokenize=13a MTNT/test/test.fr-en.en < Output/hypo.sys

BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.5.1 = 18.4 59.4/29.7/16.9/10.2 (BP = 0.785 ratio = 0.805 hyp_len = 15114 ref_len = 18777)


BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.4.14 = 23.9 55.2/29.9/17.9/11.0 (BP = 1.000 ratio = 1.034 hyp_len = 19421 ref_len = 18777)

ON IWSLT

In [ ]:
!fairseq-interactive data-bin/MTNT_fr_en/ \
    --skip-invalid-size-inputs-valid-test \
    --path checkpoint/MTNT_fr_en/checkpoint_last.pt \
    --buffer-size 2000 --batch-size 128 \
    --beam 5 --remove-bpe \
    --input iwslt17.tokenized.fr-en/test.fr | tee Output/iwslt.test.fr-en.en.sys

Streaming output truncated to the last 5000 lines.
W-455	0.002	seconds
H-455	-1.4749032258987427	<unk> <unk> <unk> <unk> , then a <unk> <unk> .
D-455	-1.4749032258987427	<unk> <unk> <unk> <unk> , then a <unk> <unk> .
P-455	-0.2597 -2.0778 -1.5419 -1.8723 -3.1097 -2.2719 -0.4521 -1.7760 -1.2635 -1.4869 -0.1121
S-456	<unk> le procrastinateur , ce conflit a tendance à se terminer de la même manière à chaque fois , le laissant passer beaucoup de temps dans cette zone orange , un endroit facile et amusant , totalement hors du <unk> de la <unk> .
W-456	0.015	seconds
H-456	-0.6996342539787292	<unk> the <unk> <unk> , this conflict has a tendency to end up in the same way every time , leaving a lot of time in this orange zone , an easy place and funny , completely outside of the <unk> <unk> <unk> .
D-456	-0.6996342539787292	<unk> the <unk> <unk> , this conflict has a tendency to end up in the same way every time , leaving a lot of time in this orange zone , an easy place and funny , completely 

In [ ]:
!grep ^H Output/iwslt.test.fr-en.en.sys | cut -f3- > Output/iwslt_hypo.tok.sys


In [ ]:
!awk '{gsub("<unk>", "");print}' Output/iwslt.test.fr-en.en.sys > Output/iwslt_hypo.tok.sys1

In [ ]:
!grep ^H Output/iwslt_hypo.tok.sys1  | cut -f3- > Output/iwslt_hypo.tok.sys


In [ ]:
!cat Output/iwslt_hypo.tok.sys | mosesdecoder/scripts/recaser/detruecase.perl | mosesdecoder/scripts/tokenizer/detokenizer.perl -l en > Output/iwslt_hypo.sys


Detokenizer Version $Revision: 4134 $
Language: en


In [ ]:
!sacrebleu --tokenize=13a iwslt17.tokenized.fr-en/test.en < Output/iwslt_hypo.sys

BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.5.1 = 13.2 59.4/28.0/15.3/8.5 (BP = 0.615 ratio = 0.673 hyp_len = 20435 ref_len = 30374)


BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.4.14 = 4.1 15.1/5.9/2.7/1.2 (BP = 1.000 ratio = 2.490 hyp_len = 75628 ref_len = 30374)


In [ ]:
!zip -r Output_finetuned.zip Output/

updating: Output/ (stored 0%)
updating: Output/hypo.tok.sys1 (stored 0%)
updating: Output/iwslt_hypo.sys (stored 0%)
updating: Output/MTNT.test.fr-en.en.sys (deflated 64%)
updating: Output/hypo.sys (stored 0%)
updating: Output/iwslt.test.fr-en.en.sys (deflated 65%)
updating: Output/hypo.tok.sys (stored 0%)
updating: Output/iwslt_hypo.tok.sys (stored 0%)


In [ ]:
!git clone 'https://github.com/neulab/compare-mt.git'

Cloning into 'compare-mt'...
remote: Enumerating objects: 1440, done.
remote: Counting objects: 100% (20/20), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 1440 (delta 4), reused 10 (delta 4), pack-reused 1420
Receiving objects: 100% (1440/1440), 6.02 MiB | 17.26 MiB/s, done.
Resolving deltas: 100% (913/913), done.


In [ ]:
cd compare-mt/

/content/nmt_tutorial/compare-mt


In [ ]:
# Requirements
!pip install -r requirements.txt
# Install the package
!python setup.py install

In [ ]:
cd ..

/content/nmt_tutorial


In [ ]:
mkdir output

In [ ]:
!compare-mt --output_directory output/ iwslt17.tokenized.fr-en/test.en Output/iwslt_hypo.sys Output/iwslt_hypo.tok.sys1


********************** Aggregate Scores ************************
BLEU:
	sys1	sys2
BLEU	7.2972	0.8326

********************** Aggregate Scores ************************
length ratio:
	sys1	sys2
length ratio	0.6068 (ref=28156, out=17085)	4.4386 (ref=28156, out=124974)

Reading frequency from the reference
********************** Word Accuracy Analysis ************************
--- word fmeas by frequency bucket
frequency	sys1	sys2
<1	0.0000	0.0000
1	0.1254	0.0000
2	0.2021	0.0013
3	0.3019	0.0000
4	0.2660	0.0000
[5,10)	0.3334	0.0007
[10,100)	0.4021	0.0137
[100,1000)	0.4862	0.0825
>=1000	0.3146	0.3914

********************** Sentence Bucket Analysis ************************
--- bucket type: length, statistic type: BLEU
length	sys1	sys2
<10	1.6400	5.5221
[10,20)	6.2127	0.9432
[20,30)	7.4931	0.8943
[30,40)	9.1757	3.8508
[40,50)	10.0284	2.4399
[50,60)	6.6500	2.3365
>=60	6.7275	0.5949

********************** Sentence Bucket Analysis ************************
--- bucket type: len(output)-len(referen

In [ ]:
!compare-mt --output_directory output/ MTNT/test/test.fr-en.en Output/hypo.sys Output/hypo.tok.sys1


********************** Aggregate Scores ************************
BLEU:
	sys1	sys2
BLEU	13.3492	8.4832

********************** Aggregate Scores ************************
length ratio:
	sys1	sys2
length ratio	0.7855 (ref=16127, out=12667)	1.4069 (ref=16127, out=22689)

Reading frequency from the reference
********************** Word Accuracy Analysis ************************
--- word fmeas by frequency bucket
frequency	sys1	sys2
<1	0.0000	0.0000
1	0.2225	0.1158
2	0.3634	0.2430
3	0.3673	0.2989
4	0.3504	0.2010
[5,10)	0.4020	0.4173
[10,100)	0.4728	0.2657
[100,1000)	0.6193	0.6306
>=1000	0.0000	0.0000

********************** Sentence Bucket Analysis ************************
--- bucket type: length, statistic type: BLEU
length	sys1	sys2
<10	8.7588	4.5759
[10,20)	13.2116	8.2627
[20,30)	13.5620	8.7545
[30,40)	15.3100	10.3736
[40,50)	16.5578	12.1944
[50,60)	13.0238	9.4530
>=60	13.6147	8.0170

********************** Sentence Bucket Analysis ************************
--- bucket type: len(output)-len(